src/seedu/addressbook/commands/Command.java

```diff
@@ -54,6 +54,13 @@ public void setData(AddressBook addressBook, List<? extends ReadOnlyPerson> rele
         this.relevantPersons = relevantPersons;
     }
 
+    /**
+     *  @return true if command types mutate the data.
+     */
+     public boolean isMutating() {
+         return true;
```

Comment:
Having a default implementation for the isMutating method helps to reduce the amount of code. However, some may argue that choosing false (or true) as the default is arbitrary and hurts the intuitiveness of the code. Making the method abstract forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.